src/package.rs

```diff
@@ -222,60 +289,57 @@ impl Package {
                                 else if let Ok(c) = get(self.download_dir()) { Some(c) }
                                 else { None };
         if let Some(c) = c {
-            if let Ok(n) = ConfigFile::parse(&c, crate::config_file::ConfigType::JSON) {
+            if let Ok(n) = ConfigFile::parse(&c, crate::config_file::ConfigType::JSON).await {
                 return Ok(n);
             }
         }
         ConfigFile::parse(
-            &ureq::get(&format!(
+            &reqwest::get(&format!(
                 "https://cdn.jsdelivr.net/npm/{}@{}/package.json",
                 self.name, self.version,
             ))
-            .call()
+            .await
             .map_err(|_| {
                 anyhow!("Request to cdn.jsdelivr.net failed, package/version doesnt exist")
             })?
-            .into_string()?,
+            .text()
+            .await?,
             crate::config_file::ConfigType::JSON,
         )
+        .await
     }
 
     /// Gets the package manifest and puts it in `self.manfiest`.
-    fn get_manifest(&mut self) {
-        let resp = ureq::get(&format!("{REGISTRY}/{}/{}", self.name, self.version))
-            .call()
-            .expect("Getting the package manifest file should not fail")
-            .into_string()
-            .expect("The package manifest file should be valid text");
+    pub async fn get_manifest(&mut self) -> Result<&Manifest> {
+        if !self.manifest.is_none() {
+            return Ok(self.manifest.as_ref().unwrap());
+        }
+        let resp = reqwest::get(&format!("{REGISTRY}/{}/{}", self.name, self.version))
+            .await?
+            .text()
+            .await?;
         if resp == "\"Not Found\"" {
-            panic!("Package {}@{} was not found", self.name, self.version)
+            return Err(anyhow!(
+                "Package {}@{} was not found",
+                self.name,
+                self.version
+            ));
         } else if resp == format!("\"version not found: {}\"", self.version) {
-            panic!(
+            return Err(anyhow!(
                 "Package {} exists, but version '{}' not found",
-                self.name, self.version
-            )
-        }
-        let _ = serde_json::from_str::<JValue>(&resp).expect("Manifest should be valid JSON");
-        self.manifest = resp
-    }
-
-    /// Gets the package tarball.
-    pub fn get_tarball(&mut self) -> Option<String> {
-        if self.manifest.is_empty() {
-            self.get_manifest();
+                self.name,
+                self.version
+            ));
         }
-        let j = serde_json::from_str::<JValue>(&self.manifest).unwrap();
-        Some(j.get("dist")?.get("tarball")?.as_str()?.to_string())
-    }
-
-    /// Gets the package integrity.
-    pub fn get_integrity(&mut self) -> Option<String> {
-        if self.manifest.is_empty() {
-            self.get_manifest();
+        #[derive(Deserialize)]
+        struct W {
```

Comment:
What is the point of this struct? Shouldn't `Deserialize` be derived on the `Manifest` directly?

Reply:
it is, but i need to get out the dist field. i suppose i could `serde_json::from_str::<JValue>(&self.manifest.unwrap()).get("dist")?.into::<Manifest>()` but im not sure if thats better..